src/plugin.ts

```diff
@@ -2,47 +2,52 @@ import { Octokit } from "@octokit/rest";
 import { Env, PluginInputs } from "./types";
 import { Context } from "./types";
 import { isIssueCommentEvent } from "./types/typeguards";
-import { helloWorld } from "./handlers/hello-world";
 import { LogLevel, Logs } from "@ubiquity-dao/ubiquibot-logger";
+import { createAdapters } from "./adapters";
+import { createClient } from "@supabase/supabase-js";
+import { addComments } from "./handlers/add-comments";
+import { updateComment } from "./handlers/update-comments";
+import { deleteComment } from "./handlers/delete-comments";
+import OpenAI from "openai";
 
 /**
  * The main plugin function. Split for easier testing.
  */
 export async function runPlugin(context: Context) {
   const { logger, eventName } = context;
-
   if (isIssueCommentEvent(context)) {
-    return await helloWorld(context);
+    switch (eventName) {
+      case "issue_comment.created":
+        return await addComments(context);
+      case "issue_comment.deleted":
+        return await deleteComment(context);
+      case "issue_comment.edited":
+        return await updateComment(context);
+      default:
+        logger.error(`Unsupported event: ${eventName}`);
+    }
   }
-
-  logger.error(`Unsupported event: ${eventName}`);
+  logger.ok(`Exiting plugin`);
 }
 
 /**
  * How a worker executes the plugin.
  */
 export async function plugin(inputs: PluginInputs, env: Env) {
   const octokit = new Octokit({ auth: inputs.authToken });
-
+  const supabase = createClient(env.SUPABASE_URL, env.SUPABASE_KEY);
+  const openaiClient = new OpenAI({
+    apiKey: env.OPENAI_API_KEY,
```

Comment:
Prefer to check and throw error if key is empty first.

Reply:
We are validating the schema before running the plugin. I think this would be redundant